src/pyfreedb/row/models.py

```diff
@@ -36,14 +36,32 @@ def __get__(self, obj: Any, _: Any) -> Optional[T]:
 
     def __set__(self, obj: Any, value: Optional[T]) -> None:
         self.__ensure_type(value)
+        if value is not NotSet:
+            value = self._typ(value)  # type: ignore [call-arg]
```

Comment:
What does this do actually? Type casting to type `T`?

Reply:
And also why need to type cast again if L52 below already has the `isinstance` check?

Reply:
This is because the `IntegerField` and `FloatField` both can accept int/float.

When we set the value to the field we're going to typecast them to the actual field's type.